dev-docs/Action Flows.md

```diff
@@ -0,0 +1,349 @@
+  "issuer": "https://example.edu/issuers/565049",
+  "issuanceDate": "2010-01-01T00:00:00Z",
+  "credentialSubject": {
+    "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
+    "degree": {
+      "type": "BachelorDegree",
+      "name": "Bachelor of Science and Arts"
+    }
+  }
+}
+```
+
+```mermaid
+flowchart LR;
+  UserAccount --> Credential;
+  TemplateStore --> Credential;
+```
+
+The issuer then signs this generated VC with its private key from the local keystore and adds it to the ```proof``` field of the JSON Credential.
+
+```json
+{
+  "@context": [
+    "https://www.w3.org/2018/credentials/v1",
+    "https://www.w3.org/2018/credentials/examples/v1",
+    "https://w3id.org/security/suites/ed25519-2020/v1"
+  ],
+  "id": "http://example.edu/credentials/3732",
+  "type": [
+    "VerifiableCredential",
+    "UniversityDegreeCredential"
+  ],
+  "issuer": "https://example.edu/issuers/565049",
+  "issuanceDate": "2010-01-01T00:00:00Z",
+  "credentialSubject": {
+    "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
+    "degree": {
+      "type": "BachelorDegree",
+      "name": "Bachelor of Science and Arts"
+    }
+  },
+  "proof": {
+    "type": "Ed25519Signature2020",
+    "created": "2022-02-25T14:58:42Z",
+    "verificationMethod": "https://example.edu/issuers/565049#key-1",
+    "proofPurpose": "assertionMethod",
+    "proofValue": "z3FXQjecWufY46yg5abdVZsXqLhxhueuSoZgNSARiKBk9czhSePTFehP
+8c3PGfb6a22gkfUKods5D2UAUL5n2Brbx"
+  }
+}
+```
+
+```mermaid
+flowchart LR;
+  UserAccount --> Credential;
+  TemplateStore --> Credential;
+  Credential --> |signed|VC;
+  VC --> CommunicationMethod;
+  CommunicationMethod --> PublicRegistry
+  CommunicationMethod --> Hooks/Plugins
+```
+
+This generated VC is then transferred to the holder via a communication channel. There are 2 major channels which can be used in this case
+
+- **Hooks / Plugins :** We can send the generated credential via a medium such as Email, WhatsApp, Telegram, etc. The Holder can download the file, and load it into the application. The application automatically saves the credential and looks for updates.
+
+  - **Note :** Every credential created using a hook must have an updateLink which should enable queries of the form ```GET /updates?id={id}``` in order to check for updates. In case such as link is not available, the app will not search for any updates related to the given VC.
+
+- **Public Registry :** The credential can also be saved on a public registry / database against the public ID of the user. This can be thought of as a ```PubSub model``` where credentials are published to the registry, and the users (who are subscribers to the registry) periodically check for new updates.
+
+  - **Note :** The app regularly checks for any new VC or any updated VC against the publicID of the holder. Updates are checked by comparing the previous and latest signatures. If they are different, it indicates a change in data, and hence the new VC is automatically downloaded.
```

Comment:
A point to note: how would this architecture support being able to provide "rememberable" identifiers to individuals?